crates/types/src/rpc/transactions.rs

```diff
@@ -326,33 +390,41 @@ pub struct Reversion {
     pub revert_reason: String,
 }
 
-#[derive(Debug, Clone, Serialize)]
+#[derive(Debug, Clone, Serialize, Deserialize)]
 #[serde(untagged)]
 pub enum ExecutionInvocation {
     Succeeded(FunctionInvocation),
     Reverted(Reversion),
 }
 
-#[derive(Debug, Clone, Serialize)]
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(deny_unknown_fields)]
 pub struct InvokeTransactionTrace {
     pub validate_invocation: Option<FunctionInvocation>,
-    pub execution_invocation: ExecutionInvocation,
+    pub execute_invocation: ExecutionInvocation,
+    pub fee_transfer_invocation: Option<FunctionInvocation>,
+    pub state_diff: ThinStateDiff,
 }
 
-#[derive(Debug, Clone, Serialize)]
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(deny_unknown_fields)]
 pub struct DeclareTransactionTrace {
     pub validate_invocation: Option<FunctionInvocation>,
     pub fee_transfer_invocation: Option<FunctionInvocation>,
+    pub state_diff: ThinStateDiff,
 }
 
-#[derive(Debug, Clone, Serialize)]
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(deny_unknown_fields)]
 pub struct DeployAccountTransactionTrace {
     pub validate_invocation: Option<FunctionInvocation>,
     pub constructor_invocation: Option<FunctionInvocation>,
     pub fee_transfer_invocation: Option<FunctionInvocation>,
+    pub state_diff: ThinStateDiff,
 }
```

Comment:
Please add `type` field to transaction trace structs 🙏🏻

Reply:
its added in  `TransactionTrace` struct #[serde(tag = "type")]